core/src/main/java/org/infinispan/partitionhandling/impl/PartitionHandlingInterceptor.java

```diff
@@ -172,6 +185,37 @@ private Object handleDataReadCommand(InvocationContext ctx, DataCommand command)
          }
 
          postOperationPartitionCheck(rCtx, dataCommand, dataCommand.getKey());
+         return rv;
+      });
+   }
+
+   private CompletableFuture<Void> handleMissingOwner(Collection<?> keys, MissingOwnerException moe) throws InterruptedException {
+      // In scattered cache mode it is possible that the CH does not contain owner of the key
```

Comment:
TODO: check in which test this happens, availiability mode always arrives with topology update